sz.py

```diff
@@ -1,28 +1,54 @@
 #!/usr/bin/env python3
-import os
-from pathlib import Path
+import os, sys
 import token
 import tokenize
 import itertools
 from tabulate import tabulate
 
 TOKEN_WHITELIST = [token.OP, token.NAME, token.NUMBER, token.STRING]
 
-if __name__ == "__main__":
-  headers = ["Name", "Lines", "Tokens/Line"]
+def gen_stats(base_path="."):
   table = []
-  for path, subdirs, files in os.walk("tinygrad"):
+  for path, _, files in os.walk(os.path.join(base_path, "tinygrad")):
     for name in files:
       if not name.endswith(".py"): continue
-      filepath = Path(path) / name
+      filepath = os.path.join(path, name)
+      relfilepath = os.path.relpath(filepath, base_path)
       with tokenize.open(filepath) as file_:
         tokens = [t for t in tokenize.generate_tokens(file_.readline) if t.type in TOKEN_WHITELIST]
         token_count, line_count = len(tokens), len(set([t.start[0] for t in tokens]))
-        table.append([filepath.as_posix(), line_count, token_count/line_count])
+        table.append([relfilepath, line_count, token_count/line_count])
+  return table
+
+def display_diff(diff): return "+"+str(diff) if diff > 0 else str(diff)
 
-  print(tabulate([headers] + sorted(table, key=lambda x: -x[1]), headers="firstrow", floatfmt=".1f")+"\n")
+def gen_diff(table_old, table_new):
+  table = []
+  for line_old, line_new in zip(table_old, table_new):
+    if line_new[1]-line_old[1] != 0:
```

Comment:
what happens when the exact same number of lines gets added and then deleted?

Reply:
its kind of an edge case so not that important but tinyb0t used to support that i think

Reply:
Yeah, not sure how feasible it is to do it with `sz.py` since itself doesn't track additions or deletions